applications/EigenSolversApplication/custom_solvers/eigen_direct_solver.h

```diff
@@ -166,12 +114,12 @@ class EigenDirectSolver
      */
     void PerformSolutionStep(SparseMatrixType& rA, VectorType& rX, VectorType& rB) override
     {
-        Eigen::Map<Eigen::Matrix<DataType, Eigen::Dynamic, 1> > x(rX.data().begin(), rX.size());
-        Eigen::Map<Eigen::Matrix<DataType, Eigen::Dynamic, 1> > b(rB.data().begin(), rB.size());
+        Eigen::Map<Eigen::Matrix<DataType, Eigen::Dynamic, 1>> x(rX.data().begin(), rX.size());
+        Eigen::Map<Eigen::Matrix<DataType, Eigen::Dynamic, 1>> b(rB.data().begin(), rB.size());
 
-        x = m_solver.solve(b);
+        const bool success = m_solver.Solve(b, x);
 
-        KRATOS_ERROR_IF(m_solver.info() != Eigen::Success) << "Solving failed!" << std::endl;
+        KRATOS_ERROR_IF(!success) << "Solving failed!" << std::endl;
```

Comment:
in order to further unify I suggest to introduce a method `PrintAdditionalErrorMessages` (name just a suggestion)
=> those could be overridden by the eigen but not the pardiso solvers to print the messages
```suggestion
        KRATOS_ERROR_IF(!success) << "Solving failed!" << PrintAdditionalErrorMessages() << std::endl;
```